src/main/java/ch/njol/skript/expressions/ExprRegexMatches.java

```diff
@@ -0,0 +1,96 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.regex.PatternSyntaxException;
+
+@Name("Regex Matches")
+@Description("Returns all the matches of a string using a regex pattern. Returns nothing if the pattern is invalid.")
+@Examples("set {_numbers::*} to the regex matches of {_input} using \"\\d+\"")
+@Since("INSERT VERSION")
+public class ExprRegexMatches extends SimpleExpression<String> {
+
+	static {
+		Skript.registerExpression(ExprRegexMatches.class, String.class, ExpressionType.COMBINED,
+				"[all [[of] the]|the] regex matches of %strings% (with|using) %string%");
```

Comment:
I'm really not a major fan of this pattern. I think it's much more natural to be looking for "matches of (needles) in (haystacks)" rather than "matches of (haystacks) with (needles)"

Reply:
Actually I think I've slightly misunderstood what this is actually supposed to do. I assumed it was getting matches of a group pattern from inside some strings (e.g. all (\d) in "123") but after looking at the description again I'm wondering if what you're aiming for is more like filtering any strings from the list that contain the group, in which case I *really* don't like this because it should be using the match rather than the find mode.